CustomerContactBook/Services/GroupService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using CustomerContactBook.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CustomerContactBook.Services
+{
+    public class GroupService
+    {
+        private readonly ContactBookContext _context;
+
+        public GroupService(ContactBookContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<ActionResult<IEnumerable<CustomerGroup>>> GetGroups()
+        {
+            return await _context.Groups.ToListAsync();
```

Comment:
Is it compilable? I'm not sure, because the method has signature `ActionResult<IEnumerable<CustomerGroup>>` but you returns `List<CustomerGroup>`.
So there is couple of important points:
1. `ActionResult` is only for controller level, services shouldn't work with that.
2. If service returns `CustomerGroup` (database model) it means controller operates with database models, that's bad practice. The controller should work with another models. In our case it may looks weird - you will just copy all fields of one class to another, but it gives a lot of good points:
* external caller doesn't have access to database model = it can't change data;
* it's easier to find usages / do refactoring;
* when you add new field in database it will not automatically appear in the API;
* vise versa - to add new API field you don't need to change database.

In the enterprise applications 3 model lawyers are used: database classes, service classes, controller classes. So service manipulates database classes, but always returns another class (service model). Controller get service models, but returns another class (controller model). It allows to reuse one service in the different controllers/places and returns different models. Also it's easier for refactoring.
In our project I don't want to bring all 3 lawyers, I think 2 are enough - first is controller+service and the second is database.

Reply:
Please fix at least the first point, then I will continue to review.

Reply:
I made it so services no longer return ActionResults or Database models in my latest commit

Reply:
Good job!